test/test_functions/test_function_kanji.py

```diff
@@ -0,0 +1,183 @@
+import pytest
+
+import escpos.printer as printer
+from escpos.constants import (
+    KANJI_DEFINE_USER_DEFINED,
+    KANJI_DELETE_USER_DEFINED,
+    KANJI_ENTER_KANJI_MODE,
+    KANJI_EXIT_KANJI_MODE,
+    KANJI_PRINT_MODE,
+    KANJI_SET_CHAR_STYLE,
+    KANJI_SET_ENCODING,
+    KANJI_SET_QUADRUPLE_SIZE,
+    KANJI_SET_SPACING,
+    KANJI_UNDERLINE,
+)
+
+checkerboard_kanji = (
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\xf0\xf0\xf0"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+    b"\x0f\x0f\x0f"
+)
+
+
+def test_enter_kanji_mode() -> None:
+    """should enter kanji mode."""
+    instance = printer.Dummy()
+    instance._enter_kanji_mode()
+    assert instance.output == KANJI_ENTER_KANJI_MODE
+
+
+def test_exit_kanji_mode() -> None:
+    """should exit kanji mode."""
+    instance = printer.Dummy()
+    instance._exit_kanji_mode()
+    assert instance.output == KANJI_EXIT_KANJI_MODE
+
+
```

Comment:
```suggestion

def test_kanji_text_without_encoding() -> None:
    """Test behavior when no encoding is set."""
    instance = printer.Dummy()
    with pytest.raises(ValueError):
        instance.kanji_text("Hello世界Hello")

```
Please test also the negative case when no encoding is set.